packages/discord-request/src/request.ts

```diff
@@ -0,0 +1,79 @@
+import type { ErrorBody } from "discord-error";
+import { DiscordError, isDiscordError } from "discord-error";
+import type { Client } from "./client.js";
+import { RateLimitError } from "./errors/rate-limit-error.js";
+import { RequestError } from "./errors/request-error.js";
+import { parse } from "./response.js";
+
+export async function request(
+	client: Client,
+	path: string,
+	init: RequestInit,
+	abortSignal: AbortSignal | null | undefined
+): Promise<ArrayBuffer | JSON | Response> {
+	// Setup the timeout signal
+	const signal = new AbortController();
+	const abortRequest = () => {
```

Comment:
this should realistically call clearAbort too

Reply:
Even if it's a timeout as opposed to an interval? I thought that was only important for setInterval

Reply:
Nevermind I see why now